src/test/java/com/google/sps/data/DatastoreAvailabilityDaoTest.java

```diff
@@ -0,0 +1,369 @@
+    List<Entity> entities =
+        datastore
+            .prepare(new Query("Availability").addSort("startTime", SortDirection.ASCENDING))
+            .asList(FetchOptions.Builder.withDefaults());
+    List<Availability> availabilities = new ArrayList<Availability>();
+    for (Entity entity : entities) {
+      availabilities.add(tester.entityToAvailability(entity));
+    }
+    Availability expectedAvailabilityOne =
+        Availability.create(
+            "user1@mail.com",
+            new TimeRange(
+                Instant.parse("2020-07-07T12:00:00Z"), Instant.parse("2020-07-07T12:15:00Z")),
+            availabilities.get(0).id(),
+            true);
+    Availability expectedAvailabilityTwo =
+        Availability.create(
+            "user1@mail.com",
+            new TimeRange(
+                Instant.parse("2020-07-07T15:45:00Z"), Instant.parse("2020-07-07T16:00:00Z")),
+            availabilities.get(1).id(),
+            false);
+    List<Availability> expectedAvailabilities = new ArrayList<Availability>();
+    expectedAvailabilities.add(expectedAvailabilityOne);
+    expectedAvailabilities.add(expectedAvailabilityTwo);
+    Assert.assertEquals(expectedAvailabilities, actual);
+  }
+
+  @Test
+  public void onlyGetsSpecifiedUsersAvailabilityInRange() {
+    tester.create(availabilityOne);
+    tester.create(availabilityTwo);
+    tester.create(availabilityThree);
+    List<Availability> actual =
+        tester.getInRangeForUser(
+            "user1@mail.com",
+            Instant.parse("2020-07-07T12:00:00Z").toEpochMilli(),
+            Instant.parse("2020-07-07T17:45:00Z").toEpochMilli());
+    List<Entity> entities =
+        datastore
+            .prepare(new Query("Availability").addSort("startTime", SortDirection.ASCENDING))
+            .asList(FetchOptions.Builder.withDefaults());
+    List<Availability> availabilities = new ArrayList<Availability>();
+    for (Entity entity : entities) {
+      availabilities.add(tester.entityToAvailability(entity));
+    }
+    Availability expectedAvailabilityOne =
+        Availability.create(
+            "user1@mail.com",
+            new TimeRange(
+                Instant.parse("2020-07-07T12:00:00Z"), Instant.parse("2020-07-07T12:15:00Z")),
+            availabilities.get(0).id(),
+            true);
+    Availability expectedAvailabilityTwo =
+        Availability.create(
+            "user1@mail.com",
+            new TimeRange(
+                Instant.parse("2020-07-07T15:45:00Z"), Instant.parse("2020-07-07T16:00:00Z")),
+            availabilities.get(1).id(),
+            false);
+    List<Availability> expectedAvailabilities = new ArrayList<Availability>();
+    expectedAvailabilities.add(expectedAvailabilityOne);
+    expectedAvailabilities.add(expectedAvailabilityTwo);
+    Assert.assertEquals(expectedAvailabilities, actual);
+  }
+
+  @Test
+  public void getsScheduledAvailabilitiesForUserInRange() {
```

Comment:
Takes a lot of mental effort to figure out what these tests are doing, especially since their names are pretty similar. Could you put comments above?